src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -1162,12 +1165,18 @@ public override Task<ImplementationResult> ImplementationAsync(DelegatedPosition
                 Uri = virtualDocument.Uri,
             },
         };
-        var response = await _requestInvoker.ReinvokeRequestOnServerAsync<VSInternalDocumentDiagnosticsParams, VSInternalDiagnosticReport[]?>(
-            virtualDocument.Snapshot.TextBuffer,
-            VSInternalMethods.DocumentPullDiagnosticName,
-            delegatedLanguageServerName,
-            request,
-            cancellationToken).ConfigureAwait(false);
+
+        ReinvocationResponse<VSInternalDiagnosticReport[]?>? response;
+        using (_requestInvoker.Track(nameof(_requestInvoker.ReinvokeRequestOnServerAsync), VSInternalMethods.DocumentPullDiagnosticName, delegatedLanguageServerName, correlationId))
```

Comment:
Does this now get tracked twice? This call will report the method name, server name and correlation id, and then the call to `ReinvokeRequestOnServerAsync` will report the method name, server name and no correlation id. I would expect correlation id to be a new parameter on all of the request methods on the request invoker, if this is the path we want to use moving forward.

Though thinking a bit broader, do we get value out of telemetry that doesn't have a correlation id? Perhaps in hindsight the `TelemetryReportingLSPRequestInvoker` is not the best solution here, and should be removed. This call will still track the telemetry we need for diagnostics, and if we want to track other messages, we can go and add the correlation id to their parameters as neeeded (or gone up with some generalized design, more likely).